raft/src/main/scala/zio/raft/Raft.scala

```diff
@@ -578,11 +579,15 @@ class Raft[S, A <: Command](
             (newState, response) <- stateMachine.apply(logEntry.command).toZIOWithState(appState)
             _ <- appStateRef.set(newState)
 
-            _ <- state match
-              case l: Leader =>
-                pendingCommands.complete(logEntry.index, response.asInstanceOf[logEntry.command.Response])
-              case _ => ZIO.unit
-          yield state.increaseLastApplied
+            newRaftState <- state match
+              case l: Leader[S] =>
+                for
+                  _ <- pendingCommands.complete(logEntry.index, response.asInstanceOf[logEntry.command.Response])
```

Comment:
we can make this prettier, move pendingCommands to the leader as well and change `complete` to handle both pendingReads and pendingCommands